aviary/interface/methods_for_level2.py

```diff
@@ -2361,16 +2361,30 @@ def _add_flops_landing_systems(self):
                                'traj.climb.initial_states:mass')
             self.model.connect(Mission.Takeoff.GROUND_DISTANCE,
                                'traj.climb.initial_states:range')
-            self.model.connect(Mission.Takeoff.FINAL_VELOCITY,
-                               'traj.climb.initial_states:velocity')
-            self.model.connect(Mission.Takeoff.FINAL_ALTITUDE,
-                               'traj.climb.initial_states:altitude')
+            if self.mission_method == "FLOPS":
+                self.model.connect(Mission.Takeoff.FINAL_VELOCITY,
+                                   'traj.climb.initial_states:velocity')
+                self.model.connect(Mission.Takeoff.FINAL_ALTITUDE,
+                                   'traj.climb.initial_states:altitude')
+            else:
+                pass
+                # TODO: connect this correctly
+                # mass is the most important to connect but these others should
+                # be connected as well
+                # self.model.connect(Mission.Takeoff.FINAL_VELOCITY,
+                #                 'traj.climb.initial_states:mach')
+                # self.model.connect(Mission.Takeoff.FINAL_ALTITUDE,
+                #                 'traj.climb.controls:altitude')
 
         self.model.connect('traj.descent.states:mass',
                            Mission.Landing.TOUCHDOWN_MASS, src_indices=[-1])
         # TODO: approach velocity should likely be connected
-        self.model.connect('traj.descent.states:altitude', Mission.Landing.INITIAL_ALTITUDE,
-                           src_indices=[-1])
+        if self.mission_method == "FLOPS":
+            self.model.connect('traj.descent.states:altitude', Mission.Landing.INITIAL_ALTITUDE,
+                               src_indices=[-1])
+        else:
+            self.model.connect('traj.descent.control_values:altitude', Mission.Landing.INITIAL_ALTITUDE,
```

Comment:
if the mission method is GASP, wouldn't this throw an error because altitude is a state?

Reply:
This logic branch is never reached in the case of GASP missions because the method is only called for FLOPS-based missions.